### PR TITLE
Add parameters for create categories

### PR DESCRIPTION
**Pull request by atomoc**

### CODE REVIEW

File: src/categories/create.js [hunk 1]
```diff
@@ -19,19 +19,19 @@ module.exports = function(Categories) {
...
+				bgColor: ( data.bgColor ? data.bgColor : '' ),
+				color: ( data.color ? data.color : '' ),
 				slug: slug,
-				parentCid: 0,
+				parentCid: ( data.parentCid ? data.parentCid : 0 ),
 				topic_count: 0,
 				post_count: 0,
 				disabled: 0,
-				order: order,
+				order: ( data.order ? data.order : cid ),
```
Comment (julianlam) [inline src/categories/create.js]: This overrides, line 17, and should be removed.

File: src/categories/create.js [hunk 2]
```diff
@@ -19,19 +19,19 @@ module.exports = function(Categories) {
 			var category = {
 				cid: cid,
 				name: data.name,
-				description: data.description,
-				icon: data.icon,
-				bgColor: data.bgColor,
-				color: data.color,
+				description: ( data.description ? data.description : '' ),
+				icon: ( data.icon ? data.icon : '' ),
+				bgColor: ( data.bgColor ? data.bgColor : '' ),
```
Comment (julianlam) [inline src/categories/create.js]: Revert this line please, I have alternate plans for the default. Thanks.

**Comment by julianlam** [comment]
Comments inline, otherwise looks good. 

@psychobunny, CLA signed?

**Comment by atomoc** [comment]
Thanks Julian!
What is CLA?

File: src/categories/create.js (hunk 2 above)
Comment (atomoc) [inline src/categories/create.js]: Only bgColor?

**Comment by julianlam** [comment]
https://www.clahub.com/agreements/NodeBB/NodeBB

**Comment by atomoc** [comment]
During the signing error is output: *"We're sorry, but something went
wrong"*

2015-03-09 17:40 GMT+03:00 Julian Lam <notifications@github.com>:

**Comment by atomoc** [comment]
​

2015-03-09 17:54 GMT+03:00 Сергей Савельев <sergej_savelev@eyrini.ru>:

**Comment by atomoc** [comment]
During the signing error is output: **"We're sorry, but something went
wrong"**

**Comment by atomoc** [comment]
I signed CLA. Please merge commits

**Comment by julianlam** [comment]
Hi @atomoc, [can you sign this one instead](https://docs.google.com/forms/d/1KrniXs3pwK8dMeAZnnQz2D07ne7LjJTYIYwJrpDqgnE/viewform)?

**Comment by atomoc** [comment]
>NodeBB Individual Contributor Assignment Agreement
Thank you for signing the NodeBB Individual Contributor License Agreement!

**Comment by julianlam** [comment]
Thanks :smile:

File: src/categories/create.js [hunk 3]
```diff
@@ -20,19 +20,19 @@ module.exports = function(Categories) {
 			var category = {
 				cid: cid,
 				name: data.name,
-				description: data.description,
-				icon: data.icon,
+				description: ( data.description ? data.description : '' ),
+				icon: ( data.icon ? data.icon : '' ),
 				bgColor: data.bgColor || colours[0],
-				color: data.color || colours[1],
+				color: ( data.color ? data.color : '' ),
```
Comment (julianlam) [inline src/categories/create.js]: This line needs to be reverted, as `colours[1]` is the correct fallback

File: src/controllers/admin.js [hunk 4]
```diff
@@ -135,11 +135,17 @@ adminController.categories.get = function(req, res, next) {
...
+        plugins.fireHook('filter:admin.categories.get', {req: req, res: res, category: data.category[0], privileges: data.privileges}, function(err, data) {
+			if (err) {
+				return next(err);
+			}
+            
+            res.render('admin/manage/category', {
+                category: data.category,
+                privileges: data.privileges
+            });
+        });
```
Comment (julianlam) [inline src/controllers/admin.js]: Please match four-space tab indentation here please :smile:

File: src/controllers/admin.js [hunk 5]
```diff
@@ -135,11 +135,17 @@ adminController.categories.get = function(req, res, next) {
 		if (err) {
 			return next(err);
 		}
-
-		res.render('admin/manage/category', {
-			category: data.category[0],
-			privileges: data.privileges
-		});
+        
+        plugins.fireHook('filter:admin.categories.get', {req: req, res: res, category: data.category[0], privileges: data.privileges}, function(err, data) {
```
Comment (julianlam) [inline src/controllers/admin.js]: We're retrieving one category, so this hook should be` filter:admin.category.get`

File: src/controllers/admin.js [hunk 6]
```diff
@@ -151,14 +157,20 @@ adminController.categories.getAll = function(req, res, next) {
...
+			}
+            
+			data.categories.filter(Boolean).forEach(function(category) {
+                (category.disabled ? disabled : active).push(category);
+            });
+
+            res.render('admin/manage/categories', {
+                active: active,
+                disabled: disabled
+            });
```
Comment (julianlam) [inline src/controllers/admin.js]: Indentation as well here please

**Comment by julianlam** [comment]
Added some notes, please let me know when they've been resolved, thanks!

**Comment by atomoc** [comment]
Ok

File: src/controllers/admin.js (hunk 6 above)
Comment (atomoc) [inline src/controllers/admin.js]: a sublime text 3 and brackets so good

Comment (julianlam) [inline src/controllers/admin.js]: Make sure you're not mixing spaces and tabs

**Comment by atomoc** [comment]
ready

**Pull request merged**